jupyter_elfinder/elfinder.py

```diff
@@ -1568,74 +1493,163 @@ def __checkArchivers(self):
-                e.update({mime: {"cmd": p7zip, "argc": "e -y", "ext": "tar"}})
+            if mime not in create:
+                create.update({mime: {"cmd": p7zip, "argc": "a -ttar", "ext": "tar"}})
+            if mime not in extract:
+                extract.update(
+                    {mime: {"cmd": p7zip, "argc": "extract -y", "ext": "tar"}}
+                )
 
             mime = "application/x-gzip"
-            if mime not in c:
-                c.update({mime: {"cmd": p7zip, "argc": "a -tgzip", "ext": "gz"}})
-            if mime not in e:
-                e.update({mime: {"cmd": p7zip, "argc": "e -y", "ext": "tar.gz"}})
+            if mime not in create:
+                create.update({mime: {"cmd": p7zip, "argc": "a -tgzip", "ext": "gz"}})
+            if mime not in extract:
+                extract.update(
+                    {mime: {"cmd": p7zip, "argc": "extract -y", "ext": "tar.gz"}}
+                )
 
             mime = "application/x-bzip2"
-            if mime not in c:
-                c.update({mime: {"cmd": p7zip, "argc": "a -tbzip2", "ext": "bz2"}})
-            if mime not in e:
-                e.update({mime: {"cmd": p7zip, "argc": "e -y", "ext": "tar.bz2"}})
+            if mime not in create:
+                create.update({mime: {"cmd": p7zip, "argc": "a -tbzip2", "ext": "bz2"}})
+            if mime not in extract:
+                extract.update(
+                    {mime: {"cmd": p7zip, "argc": "extract -y", "ext": "tar.bz2"}}
+                )
 
             mime = "application/zip"
-            if mime not in c:
-                c.update({mime: {"cmd": p7zip, "argc": "a -tzip", "ext": "zip"}})
-            if mime not in e:
-                e.update({mime: {"cmd": p7zip, "argc": "e -y", "ext": "zip"}})
+            if mime not in create:
+                create.update({mime: {"cmd": p7zip, "argc": "a -tzip", "ext": "zip"}})
+            if mime not in extract:
+                extract.update(
+                    {mime: {"cmd": p7zip, "argc": "extract -y", "ext": "zip"}}
+                )
 
         if not self._options["archiveMimes"]:
-            self._options["archiveMimes"] = list(c.keys())
+            self._options["archiveMimes"] = list(create.keys())
         else:
             pass
 
         self._options["archivers"] = archive
-        pass
-
-    def __runSubProcess(self, cmd, validReturn=[0]):
-        try:
-            sp = subprocess.Popen(
-                cmd,
-                shell=False,
-                stdout=subprocess.PIPE,
-                stderr=subprocess.PIPE,
-                stdin=subprocess.PIPE,
-            )
-            out, err = sp.communicate("")
-            ret = sp.returncode
-            # print cmd, ret, out, err
-        except (subprocess.SubprocessError, OSError):
-            return False
-
-        if ret not in validReturn:
-            return False
-
-        return True
 
-    def __checkUtf8(self, name):
+    def __check_utf8(self, name):
```

Comment:
@oeway Do you know what cases we need to handle here? The logic seems not really fit together. Ie, the try... except block is only activated if version is below or equal to 3.0. So then the except block where we check for version above 3.0 should not be needed, right?

Maybe we can simplify this when we remove support for Python 2 completely?

Reply:
I think we can just drop the support for python2.

BTW,

There are couple of places which I had to do :

if isinstance(name, bytes):
       name = name.decode("utf-8")

Perhaps we can wrap those into this check_utf8 function?

Reply:
Ok, I can do that in a follow up.

We just want to make sure that the data is a string instead of bytes, right?

Reply:
Yes.